src/interp/interp.cc

```diff
@@ -632,31 +632,18 @@ Result Memory::Copy(Memory& dst,
   return Result::Error;
 }
 
-Value Instance::ResolveInitExpr(Store& store, InitExpr init) {
-  Value result;
-  switch (init.kind) {
-    case InitExprKind::I32:      result.Set(init.i32_); break;
-    case InitExprKind::I64:      result.Set(init.i64_); break;
-    case InitExprKind::F32:      result.Set(init.f32_); break;
-    case InitExprKind::F64:      result.Set(init.f64_); break;
-    case InitExprKind::V128:     result.Set(init.v128_); break;
-    case InitExprKind::GlobalGet: {
-      Global::Ptr global{store, globals_[init.index_]};
-      result = global->Get();
-      break;
-    }
-    case InitExprKind::RefFunc: {
-      result.Set(funcs_[init.index_]);
-      break;
-    }
-    case InitExprKind::RefNull:
-      result.Set(Ref::Null);
-      break;
-
-    case InitExprKind::None:
-      WABT_UNREACHABLE;
+Result Instance::CallInitFunc(Store& store,
+                              const Ref func_ref,
+                              Value* result,
+                              Trap::Ptr* out_trap) {
+  Values results;
+  Func::Ptr func{store, func_ref};
+  if (Failed(func->Call(store, {}, results, out_trap))) {
```

Comment:
Our current set of operators can't trap, right? What other kinds of errors can happen?

Reply:
None that I know of.  Validation should mean that nothing interesting can happen here.  But checking the result of the Call worked is still good idea I think.

Reply:
Yeah, I guess it could be just an assert, since these functions would already have been validated.